docs(ocm): update OCM configuration documentation (#763)
janus-idp/backstage-plugins: plugins/ocm/README.md

#### Prerequisites

- OCM is deployed and configured on a Kubernetes cluster.
- [Kubernetes plugin for Backstage](https://backstage.io/docs/features/kubernetes) is installed and configured (Optional)

> - A `ClusterRole` is granted to a `ServiceAccount` with access to the hub cluster as follows:

Reviewer: ```suggestion
- On the target Hub clusters please create a `ServiceAccount` and grant it following `ClusterRole`:
```

         ocm:
           env: # Key is reflected as provider ID. Defines and claims plugin instance ownership of entities
             name: # Name that the hub cluster will assume in Backstage Catalog (in OCM this is always local-cluster which can be confusing)
             url: # URL of the hub cluster API endpoint
             serviceAccountToken: # Token used for querying data from the hub
             skipTLSVerify: # Skip TLS certificate verification, defaults to false (Optional)
             caData: # Base64-encoded CA bundle in PEM format (Optional)
     ```

>      - To ensure the cluster resources are properly ingested into the catalog, the value provided to the `name` field must only contain alphanumeric characters and `-_.`, as well as contain at most 62 characters. (Should satisfy the following regex `^[a-zA-Z0-9-_.]{1,62}$`)

Reviewer: ```suggestion
     - To ensure the Hub clusters are properly ingested into the catalog, the value provided to the `name` field must comply with https://backstage.io/docs/features/software-catalog/descriptor-format/#name-required requirements
```

>    - Alternatively, If the Backstage Kubernetes plugin is installed and configured to connect to the hub cluster, then you can bind the both hub and Kubernetes configuration by providing the name of the hub in the `app-config.yaml` as follows:

Reviewer: ```suggestion
   - Alternatively, If the Backstage Kubernetes plugin is installed and configured to connect to the Hub cluster, then you can bind the both hub and Kubernetes configuration by providing the name of the Hub in the `app-config.yaml` as follows:
```

       clusterLocatorMethods:
         - type: 'config'
           clusters:
             # highlight-add-start
             - name: <cluster-name>
               # Other cluster configurations
               url: <cluster-api-url>
               serviceAccountToken: <token>
               skipTLSVerify: # Skip TLS certificate verification, defaults to false (Optional)
               caData: # Base64-encoded CA bundle in PEM format (Optional)
             # highlight-add-end

     catalog:
       providers:
         ocm:
           env: # Key is reflected as provider ID. Defines and claims plugin instance ownership of entities
             # highlight-add-start
             kubernetesPluginRef: <cluster-name> # Match the cluster name in kubernetes plugin config
             # The remaining OCM configurations are ignored if kubernetesPluginRef is provided (with exception of schedule and owner)
             # ... Other OCM Configurations
             # highlight-add-end
     ```

>      - If the `kubernetesPluginRef` is provided, then it will take precedence over the configurations in the ocm providers

Reviewer: ```suggestion
     - If the `kubernetesPluginRef` is provided, then it will take precedence over any other configuration for that OCM provider
```
